src/main/java/org/jvnet/hudson/test/HudsonTestCase.java

```diff
@@ -1592,11 +1593,25 @@ public void run() {
             return r.get(0);
         }
 
-        public HtmlPage search(String q) throws IOException, SAXException {
+        public HtmlPage search(String q) throws IOException, SAXException, InterruptedException {
             HtmlPage top = goTo("");
-            HtmlForm search = top.getFormByName("search");
-            search.getInputByName("q").setValue(q);
-            return (HtmlPage)HtmlFormUtil.submit(search, null);
+            HtmlButton button = top.querySelector("#button-open-command-palette");
+
+            // Legacy versions of Jenkins
+            if (button == null) {
+                HtmlForm search = top.getFormByName("search");
+                search.getInputByName("q").setValue(q);
+                return (HtmlPage)HtmlFormUtil.submit(search, null);
+            }
+
+            button.click();
+
+            HtmlInput search = top.querySelector("#command-bar");
+            search.setValue(q);
+
+            Thread.sleep(100);
```

Comment:
would be good if we could avoid this sleep, is there something like seleniums expectedconditions available? 

this may introduce flakiness if it takes longer due to something like running lots of tests or a slow machine

Reply:
Likely cause of failures in https://ci.jenkins.io/blue/organizations/jenkins/Core%2Fjenkins/detail/PR-7569/57/tests/

e.g. 

```
Error
Cannot invoke "org.htmlunit.html.HtmlLink.click()" because "firstResult" is null
Stacktrace
java.lang.NullPointerException: Cannot invoke "org.htmlunit.html.HtmlLink.click()" because "firstResult" is null
	at org.jvnet.hudson.test.JenkinsRule$WebClient.search(JenkinsRule.java:2647)
	at hudson.search.SearchTest.testFailure(SearchTest.java:74)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:656)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

Reply:
although looking more I think its that this function doesn't handle it when the test is trying to have no results

Reply:
(not convinced this works at all)